soroban-env-host/src/host.rs

```diff
@@ -948,7 +979,12 @@ impl Host {
                     let closure = AssertUnwindSafe(move || cfs.call(&func, self, args));
                     let res: Result<Option<RawVal>, PanicVal> = testutils::call_with_suppressed_panic_hook(closure);
                     match res {
-                        Ok(Some(rawval)) => Ok(rawval),
+                        Ok(Some(rawval)) => {
+                            // This is under testutils, so no need to check is_debug since it's
+                            // already done in fn_return_diagnostics.
+                            self.fn_return_diagnostics(id, &func, &rawval)?;
```

Comment:
Nit: I guess the `self.is_debug()` check everywhere else is also not needed since you are already checking it inside the function?

Reply:
So I wanted this check on the outside as well because I originally had a conversion into the function which I didn't want run in non debug mode, but that's no longer true for most of these calls. The `fn_call_diagnostics` call in `vm` does a `Symbol::from_str(func)` call though so I think that should be wrapped, and if that's wrapped maybe we keep the rest wrapped for consistency? What do you think?

Reply:
Yeah, makes sense. It's also doesn't hurt to have the extra layer of check.